NewFailoverClient seems to not support TLS

`NewFailoverClient` seems to not support TLS, a tcpdump shows plaintext auth.

## Expected Behavior

It should PING the Redis server.

## Current Behavior

```
$ go run .
redis: 2021/04/20 10:11:24 sentinel.go:682: sentinel: discovered new sentinel="r1.example.org:26379" for master="master"
redis: 2021/04/20 10:11:24 sentinel.go:643: sentinel: new master="master" addr="r0.example.org:6379"
panic: read: connection reset by peer
```

## Steps to Reproduce

```go
package main

import (
	"context"
	"crypto/tls"
	"fmt"
	"time"

	"github.com/go-redis/redis/v8"
)

func main() {
	rdb := redis.NewFailoverClient(&redis.FailoverOptions{
		MasterName: "master",
		SentinelAddrs: []string{
			"r0.example.org:26379",
			"r1.example.org:26379",
		},
		SentinelPassword: "strongpassword",
		Password:         "strongpassword",
		TLSConfig:        &tls.Config{},
		DB:               1,
	})

	err := rdb.Ping(context.Background()).Err()
	if err != nil {
		panic(err)
	}
}
```

## Context (Environment)

- `github.com/go-redis/redis/v8 v8.8.2`
- `redis_version:6.2.1`

<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->

<!--- Provide a general summary of the issue in the Title above -->

## Detailed Description

<!--- Provide a detailed description of the change or addition you are proposing -->

## Possible Implementation

<!--- Not obligatory, but suggest an idea for implementing addition or change -->

--- a/sentinel.go
+++ b/sentinel.go
@@ -219,14 +219,21 @@ func masterSlaveDialer(
 				failover.trySwitchMaster(ctx, addr)
 			}
 		}
-
 		if err != nil {
 			return nil, err
 		}
 		if failover.opt.Dialer != nil {
 			return failover.opt.Dialer(ctx, network, addr)
 		}
-		return net.DialTimeout("tcp", addr, failover.opt.DialTimeout)
+
+		netDialer := &net.Dialer{
+			Timeout:   failover.opt.DialTimeout,
+			KeepAlive: 5 * time.Minute,
+		}
+		if failover.opt.TLSConfig == nil {
+			return netDialer.DialContext(ctx, network, addr)
+		}
+		return tls.DialWithDialer(netDialer, network, addr, failover.opt.TLSConfig)
 	}
 }
 

